### PR TITLE
#Use std::string reference instead of char* for utils::findChild

### PR DESCRIPTION
**Pull request by halx99**

### CODE REVIEW

File: tests/cpp-tests/Resources/ccs-res [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 892e3cbdc27d24c39c62d700a81be27e593823b7
```
Comment (minggo) [inline tests/cpp-tests/Resources/ccs-res]: please don't modify the submodules

Comment (halx99) [inline tests/cpp-tests/Resources/ccs-res]: OK, I will revert its.

**Comment by sbrednikhin** [comment]
Sorry for late answer.
For me  '\0' == name[0] looks less clear than strcmp. And using std::string reference can be uncomfortable to use (and possibly breaks backward capability for users, who will use this method from 3.12 revision).

**Comment by halx99** [comment]
What's about 0 == strlen(name)?
Why uncomfortable, const std::string& parameter type can accept std::string or c style string.

**Comment by minggo** [comment]
yep, it will not break compatibility, that's why i merge it.

**Comment by sbrednikhin** [comment]
I'v checked. Looks like you absolutely right. My bad. 
Yes, this variant is much better.

**Pull request merged**